client/my-sites/post-selector/selector.jsx

```diff
@@ -85,24 +89,31 @@ const PostSelectorPosts = React.createClass( {
 		}, SEARCH_DEBOUNCE_TIME_MS );
 	},
 
+	componentWillReceiveProps( nextProps ) {
+		if ( ! isEqual( this.props.query, nextProps.query ) ) {
+			this.setState( {
+				requestedPages: [ 1 ]
```

Comment:
Based upon the logic in `onSearch` here it looks like `this.state.searchTerm` _should_ always be the same as the value coming in via the query.  What about adding a check for equality of `this.props.siteId !== nextProps.siteId`